tests/unit/stall_detector_test.cc

```diff
@@ -28,43 +28,75 @@
 #include <seastar/testing/thread_test_case.hh>
 #include <atomic>
 #include <chrono>
+#include <sys/mman.h>
 
 using namespace seastar;
 using namespace std::chrono_literals;
 
+static seastar::logger testlog("testlog");
+
 class temporary_stall_detector_settings {
     std::chrono::milliseconds _old_threshold;
     std::function<void ()> _old_report;
 public:
+    /**
+     * Temporarily (until destructor) overload the stall detector threshold and reporting function.
+     *
+     * Also resets the reported stalls counter to zero, so the next backtraces will not be supressed.
+     */
     temporary_stall_detector_settings(std::chrono::duration<double> threshold, std::function<void ()> report)
             : _old_threshold(engine().get_blocked_reactor_notify_ms())
             , _old_report(engine().get_stall_detector_report_function()) {
         engine().update_blocked_reactor_notify_ms(std::chrono::duration_cast<std::chrono::milliseconds>(threshold));
         engine().set_stall_detector_report_function(std::move(report));
     }
+    /**
+     * Temporarily (until destructor) overload the stall detector threshold and reported stalls counter,
+     * leaving reporting function alone.
+     */
+    temporary_stall_detector_settings(std::chrono::duration<double> threshold)
+            : temporary_stall_detector_settings(threshold, engine().get_stall_detector_report_function()) {}
+
     ~temporary_stall_detector_settings() {
         engine().update_blocked_reactor_notify_ms(_old_threshold);
         engine().set_stall_detector_report_function(std::move(_old_report));
     }
 };
 
-void spin(std::chrono::duration<double> how_much) {
+template <typename B>
+void spin(std::chrono::duration<double> how_much, B body) {
```

Comment:
It's customary to use SEASTAR_CONCEPT(constaint<B>) to tell the user what to expect here. Or even use std::function<>, if the body is slow enough to dominate over the virtual call.

Reply:
I've changed it to use `std::function`. Perf doens't matter much because the kernel spin mode takes much longer to run the body (2x m[un]maps) than a virtual call and for the userspace loop it doens't matter either way since the function invoke and the body are both in userspace which is what matters.